### PR TITLE
Replace Chicker fixture with Dribbler Damper in PhysicsRobot

### LINKED ISSUE #1695: Remove chicker object from PhysicsRobot
### Description of the task

A PhysicsRobot is currently made up of 3 main components
* The body
* The chicker
* The dribbler

The chicker is a very thin sliver that sits at the back of the dribbler and exists for the purpose of triggering autokick/chip callbacks. We don't really need an entire extra fixture just to trigger specific callbacks, which can reasonably be triggered by the dribbler. There's no real benefit to having the trigger point a cm or so further back in the robot. This is technically slightly closer to where the breakbeam will activate in real life, but it's a pretty small difference.

Removing the chicker and combining it's callback functionality with the dribbler fixture would remove some code complexity from the PhysicsRobot code (mostly in setting up fixtures) and ContactListener.

### Acceptance criteria

- [ ] Remove the chicker fixture from PhysicsRobot and combine its functionality with the dribbler

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Removed the chicker from the physics robot and moved its functionality to the dribbler. Effectively renamed the old chicker fixture to be the new "dribbler damper" that damps incoming balls so they can be caught and dribbled by the robot.

### Testing Done

Unit tests pass. Functionality is equivalent

### Resolved Issues

resolves #1695 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Looks like registerDribblerDamperBallStartContactCallback isn't defined anywhere

File: src/software/simulation/physics/physics_robot.cpp [hunk 1]
```diff
@@ -16,12 +16,12 @@ extern "C"
...
-// chicker just needs to be large enough to collide with the ball and detect collisions
+// We can use a very small value for the dribbler damper thickness since the
+// damper just needs to be large enough to collide with the ball and detect collisions
 // without letting the ball tunnel and collide with the robot body
-const double PhysicsRobot::chicker_thickness = 0.005;
-const double PhysicsRobot::total_chicker_depth =
-    PhysicsRobot::dribbler_depth + PhysicsRobot::chicker_thickness;
+const double PhysicsRobot::dribbler_damper_thickness = 0.005;
+const double PhysicsRobot::total_dribbler_depth =
+    PhysicsRobot::dribbler_depth + PhysicsRobot::dribbler_damper_thickness;
```
[the remark this answers is not in the document]
Comment (MathewMacDougall) [reply src/software/simulation/physics/physics_robot.cpp]: :shrug: fixed

File: src/software/simulation/physics/simulation_contact_listener.h [hunk 2]
```diff
@@ -25,17 +25,18 @@ class SimulationContactListener : public b2ContactListener
     /**
      * Given the user data of 2 objects involved in a contact, returns a pair of
      * pointers to the physics objects involved in the contact if there was a contact
-     * between a ball and robot chicker. Otherwise returns std::nullopt
+     * between a ball and dribbler damper. Otherwise returns std::nullopt
```
Comment (kelvinkoon) [inline src/software/simulation/physics/simulation_contact_listener.h]: "if there was a contact **point**" sounds better.

File: src/software/simulation/physics/physics_robot_model.h [hunk 3]
```diff
@@ -17,11 +17,11 @@ class PhysicsRobotModel
      * Together these functions return 3 polygons that together make up the shape of the
      * robot. It is broken up this way because Box2D can only have convex polygons. We
      * split the body into 3 parts:
-     * - The main body, which is everything behind the chicker
-     * - The front-left part, which is the bit that is to the front-left of the chicker,
-     * partially enclosing it
-     * - The front-right part, which is the bit that is to the front-right of the chicker,
+     * - The main body, which is everything behind the dribbler
+     * - The front-left part, which is the bit that is to the front-left of the dribbler,
```
Comment (kelvinkoon) [inline src/software/simulation/physics/physics_robot_model.h]: Remove "that is" to make it more concise ("which is the bit to the front-left...")

File: src/software/simulation/physics/physics_robot_model.h [hunk 4]
```diff
@@ -17,11 +17,11 @@ class PhysicsRobotModel
...
      * partially enclosing it
+     * - The front-right part, which is the bit that is to the front-right of the
```
Comment (kelvinkoon) [inline src/software/simulation/physics/physics_robot_model.h]: remove "that is" to make it more concise ("which is the bit to the front-right...")

File: src/software/simulation/physics/simulation_contact_listener.h [hunk 5]
```diff
@@ -25,17 +25,18 @@ class SimulationContactListener : public b2ContactListener
...
      *
      * @param user_data_a The user data for the first object in the contact
      * @param user_data_b The user data for the second object in the contact
      *
      * @return A pair of pointers to the physics objects involved in the constact if there
-     * was a contact between a ball and robot chicker, and returns std::nullopt
+     * was a contact between a ball and dribbler damper, and returns std::nullopt
```
Comment (kelvinkoon) [inline src/software/simulation/physics/simulation_contact_listener.h]: Same here. "A contact" sounds strange.

File: src/software/simulation/simulator_robot.cpp [hunk 6]
```diff
@@ -464,3 +420,45 @@ void SimulatorRobot::runCurrentPrimitive(std::shared_ptr<FirmwareWorld_t> firmwa
...
+
+    Point dribble_point =
+        robot.position() +
+        Vector::createFromAngle(robot.orientation())
+            .normalize(DIST_TO_FRONT_OF_ROBOT_METERS + BALL_MAX_RADIUS_METERS -
+                       PhysicsRobot::DRIBBLER_DEPTH);
+    Vector dribble_force_vector = dribble_point - ball.position();
+    // convert to cm so we operate on a small scale
+    double dist_from_dribble_point_cm =
+        dribble_force_vector.length() * CENTIMETERS_PER_METER;
```
Comment (kelvinkoon) [inline src/software/simulation/simulator_robot.cpp:445]: Can we add a newline here? Almost thought this was part of the big comment below.

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
LGTM! Very minor nits.

File: src/software/simulation/physics/physics_robot_model.h (hunk 3 above)
Comment (MathewMacDougall) [reply src/software/simulation/physics/physics_robot_model.h]: done

File: src/software/simulation/physics/physics_robot_model.h (hunk 4 above)
Comment (MathewMacDougall) [reply src/software/simulation/physics/physics_robot_model.h]: done

File: src/software/simulation/physics/simulation_contact_listener.h (hunk 5 above)
Comment (MathewMacDougall) [reply src/software/simulation/physics/simulation_contact_listener.h]: fixed

File: src/software/simulation/simulator_robot.cpp (hunk 6 above)
Comment (MathewMacDougall) [reply src/software/simulation/simulator_robot.cpp:445]: done

**Review by LiCody — APPROVED** [review]
LGTM 👍

**Review by jonathanlew — APPROVED** [review]
:+1:

**Comment by MathewMacDougall** [comment]
@kelvinkoon need your review

**Review by jared-paul — APPROVED** [review]
LGTM!

**Review by kelvinkoon — APPROVED** [review]

**Pull request merged**